internal/controller/storageclient_controller.go

```diff
@@ -438,6 +439,9 @@ func (r *storageClientReconcile) deleteStorageClaims(claimName string) error {
 	storageClaim.Name = claimName
 
 	if err := r.Client.Get(r.ctx, types.NamespacedName{Name: claimName}, storageClaim); client.IgnoreNotFound(err) != nil {
+		if meta.IsNoMatchError(err) {
```

Comment:
```suggestion
	if err := r.Client.Get(r.ctx, types.NamespacedName{Name: claimName}, storageClaim); err != nil {
		if meta.IsNoMatchError(err) || kerrors.IsNotFound(err) {
```
and also remove the UID check.